File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

let button = document.getElementById('text-button');


const appRoom = consumer.subscriptions.create("RoomChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    const message = document.getElementById('message');
    message.insertAdjacentHTML('beforeend', data["message"]);
    addMessageClass();
  },

  speak: function(message) {
    let current_user_id = $("#current_user_id").val();
    let room_id = $("#room_id").val();
    return this.perform('speak',{
      message: message,
      current_user_id: current_user_id,
      room_id, room_id,
    });
  }
});
window.addEventListener("keydown", function(e) {
  if (e.key === 'Enter') {
    appRoom.speak(e.target.value);
    e.target.value='';
    e.preventDefault();
  }
})

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    Chat.create!(
      room_id: data['room_id'].to_i,
      user_id: data['current_user_id'].to_i,
      message: data['message']
    )
  end
end
# data['room_id'].to_i,data['current_user_id'].to_i
